### PR TITLE
Integrate the render templates route

### PR DESCRIPTION
**Pull request by oleh-momot**

### CODE REVIEW

File: registry/server/templates/services/renderTemplate.ts [hunk 1]
```diff
@@ -71,18 +65,87 @@ function matchIncludesAttributes(template: any) {
...
+
+    if (!includes.length) {
+        return [];
+    }
+
+    return Promise.all(includes.map(async (include: string) => {
+        try {
+            const {
+                src,
+                timeout = 10000,
```
Comment (StyleT) [inline registry/server/templates/services/renderTemplate.ts]: Let's make `3000` by default

File: registry/server/templates/services/renderTemplate.ts [hunk 2]
```diff
@@ -71,18 +65,87 @@ function matchIncludesAttributes(template: any) {
...
+                timeout: +timeout,
+            });
+
+            const stylesheets = selectStylesheets(link);
+
+            if (!stylesheets.length) {
+                return wrapWithComments(id, data);
+            }
+
+            return wrapWithComments(id, data + '\n' + stylesheets.join('\n'));
```
Comment (StyleT) [inline registry/server/templates/services/renderTemplate.ts]: stylesheets should be above markup

File: registry/server/templates/services/renderTemplate.ts [hunk 3]
```diff
@@ -71,18 +65,87 @@ function matchIncludesAttributes(template: any) {
...
+            console.error(error);
+            return include;
+        }
+    }));
+}
+
+function selectStylesheets(link: string): Array<string> {
+    const includeLinkHeader = parseLinkHeader(link);
+
+    if (!includeLinkHeader.length) {
```
Comment (StyleT) [inline registry/server/templates/services/renderTemplate.ts]: I don't think you need this code

File: registry/server/templates/services/renderTemplate.ts [hunk 4]
```diff
@@ -71,18 +65,87 @@ function matchIncludesAttributes(template: any) {
...
+        }
+
+        stylesheets.push(`<link rel="stylesheet" href="${attributes.uri}">`);
+
+        return stylesheets;
+    }, []);
+}
+
+function wrapWithComments(id: string, data: string): string {
+    return `<!-- Include "${id}" START -->\n` + data + `\n<!-- Include "${id}" END -->`;
```
Comment (StyleT) [inline registry/server/templates/services/renderTemplate.ts]: Could you rename it to `Template include "${id}" START`?

Comment (StyleT) [reply registry/server/templates/services/renderTemplate.ts]: To clearly distinguish template includes from ILC ones.

File: registry/server/templates/services/renderTemplate.ts [hunk 5]
```diff
@@ -1,59 +1,53 @@
...
+    return Object.keys(includesAttributes).reduce((
+        template: string,
+        include: string,
+        includeDataIndex: number,
     ) => template.split(include).join(includesData[includeDataIndex]), template);
 };
 
-function matchIncludesAttributes(template: any) {
+function matchIncludesAttributes(template: string): IncludesAttributes {
     const includesRegExp = /\<include(?:\s*\w*\=\"[^\"]*\"\s*)*\s*(?:\/\>|\>\s*.*\s*\<\/include\>)/gmi;
```
Comment (StyleT) [inline registry/server/templates/services/renderTemplate.ts]: Remove redundant escape symbols

Comment (StyleT) [reply registry/server/templates/services/renderTemplate.ts]: same in regexp below

Comment (StyleT) [reply registry/server/templates/services/renderTemplate.ts]: `"=<>`

File: registry/server/templates/services/renderTemplate.ts (hunk 2 above)
Comment (oleh-momot) [reply registry/server/templates/services/renderTemplate.ts]: Updated

File: registry/server/templates/services/renderTemplate.ts (hunk 3 above)
Comment (oleh-momot) [reply registry/server/templates/services/renderTemplate.ts]: Yeah, I agree
Removed

File: registry/server/templates/services/renderTemplate.ts (hunk 4 above)
Comment (oleh-momot) [reply registry/server/templates/services/renderTemplate.ts]: Sounds reasonable
Renamed

File: registry/server/templates/services/renderTemplate.ts (hunk 5 above)
Comment (oleh-momot) [reply registry/server/templates/services/renderTemplate.ts]: Removed

File: registry/server/templates/services/renderTemplate.ts (hunk 1 above)
Comment (oleh-momot) [reply registry/server/templates/services/renderTemplate.ts]: Discussed
We decided to leave 10s

File: registry/server/templates/services/renderTemplate.ts [hunk 6]
```diff
@@ -71,18 +65,83 @@ function matchIncludesAttributes(template: any) {
...
+                headers: {
+                    link,
+                }
+            } = await axios.get(src, {
+                timeout: +timeout,
+            });
+
+            const stylesheets = selectStylesheets(link);
+
+            if (!stylesheets.length) {
```
Comment (StyleT) [inline registry/server/templates/services/renderTemplate.ts]: Rm & rm `\n`

Comment (oleh-momot) [reply registry/server/templates/services/renderTemplate.ts]: Removed

**Pull request merged**